onnxruntime/core/optimizer/qdq_transformer/selectors_actions/shared/utils.cc

```diff
@@ -0,0 +1,155 @@
+static const OpVersionsAndSelector::OpVersionsMap GetMatMulOpVersionsMap() { return {{"MatMul", {}}}; }
+
+/* Selector Rules Related */
+void RegisterMiscSelectors(Selectors& qdq_selectors) {
+  /* register selectors for miscellaneous ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::DropDQDNodesSelector());
+  qdq_selectors.RegisterSelector(GetMiscOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterUnarySelectors(Selectors& qdq_selectors) {
+  /* regsiter selectors for unary ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::UnarySelector());
+  qdq_selectors.RegisterSelector(GetUnaryOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterBinarySelectors(Selectors& qdq_selectors) {
+  /* register selectors for binary ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::BinarySelector());
+  qdq_selectors.RegisterSelector(GetBinaryOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterVariadicSelectors(Selectors& qdq_selectors) {
+  /* register selectors for variadic ops */
+  std::unique_ptr<BaseSelector> selector(new QDQ::VariadicSelector());
+  qdq_selectors.RegisterSelector(GetVariadicOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterConvSelector(Selectors& qdq_selectors) {
+  /* register selector for conv op */
+  std::unique_ptr<BaseSelector> selector(new QDQ::ConvSelector());
+  qdq_selectors.RegisterSelector(GetConvOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void RegisterMatMulSelector(Selectors& qdq_selectors) {
+  /* register selector for matmul op */
+  std::unique_ptr<BaseSelector> selector(new QDQ::MatMulSelector());
+  qdq_selectors.RegisterSelector(GetMatMulOpVersionsMap(),
+                                 std::move(selector));
+}
+
+void SelectorManager::CreateSelectors() {
+  RegisterMiscSelectors(qdq_selectors_);
+  RegisterUnarySelectors(qdq_selectors_);
+  RegisterBinarySelectors(qdq_selectors_);
+  RegisterVariadicSelectors(qdq_selectors_);
+  RegisterConvSelector(qdq_selectors_);
+  RegisterMatMulSelector(qdq_selectors_);
+}
+
+void SelectorManager::InitializeSelectorsMap() {
+  for (const auto& entry : qdq_selectors_.SelectorsSet()) {
+    for (const auto& op_info : entry->op_versions_map) {
+      bool inserted = op_type_to_selectors_map_.insert({op_info.first, &*entry}).second;
+      ORT_ENFORCE(inserted, "Multiple entries for operator is not supported. OpType=", op_info.first);
+    }
+  }
+}
+
+void SelectorManager::Initialize() {
+  CreateSelectors();
+  InitializeSelectorsMap();
+}
+
+const std::vector<NodeGroup> SelectorManager::GetQDQSelections(const GraphViewer& graph_viewer) const {
+  std::vector<NodeGroup> qdq_selections;
+  for (auto index : graph_viewer.GetNodesInTopologicalOrder()) {
+    const auto* node = graph_viewer.GetNode(index);
+    if (node->Domain() != kOnnxDomain) {
+      break;
```

Comment:
>break [](http://example.com/codeflow?start=6&length=5)

why do we break instead of continue here? my understanding is that we want to consider each node to some extent at least

Reply:
modified this. not supposed to break here. will add a more general test case with better graph to catch up errors.